nuxeo-ai-model/src/main/java/org/nuxeo/ai/bulk/DataSetBulkAction.java

```diff
@@ -132,7 +132,7 @@ protected void compute(CoreSession coreSession, List<String> ids, Map<String, Se
             for (String id : ids) {
                 try {
                     DocumentModel doc = coreSession.getDocument(new IdRef(id));
-                    BlobTextFromDocument subDoc = PropertyUtils.docSerialize(doc, customProperties);
+                    BlobTextFromDocument subDoc = PropertyUtils.docSerialize(doc, new HashSet<>(customProperties));
```

Comment:
why do you need to wrap it around? making it mutable?

Reply:
I changed the type from List to Set